Core/include/Acts/EventData/SingleBoundTrackParameters.hpp

```diff
@@ -142,13 +142,31 @@ class SingleBoundTrackParameters {
     return m_paramSet.getUncertainty<kIndex>();
   }
 
+  /// Space-time position four-vector.
+  ///
+  /// @param[in] geoCtx Geometry context for the local-to-global transformation
+  ///
+  /// This uses the associated surface to transform the local position on the
+  /// surface to globalcoordinates. This requires a geometry context to select
+  /// the appropriate transformation and might be a computationally expensive
+  /// operation.
+  Vector4D position4(const GeometryContext& geoCtx) const {
+    const Vector2D loc(get<eBoundLoc0>(), get<eBoundLoc1>());
+    const Vector3D dir =
+        makeDirectionUnitFromPhiTheta(get<eBoundPhi>(), get<eBoundTheta>());
+    Vector4D pos4 = Vector4D::Zero();
```

Comment:
You don't need the zero init I think.

Reply:
I have had uninitialized warnings in the past and it is good practice to always initialize all variables to a known value. I hope that the compiler optimizes the initial zeroing away anyways.

Reply:
I wouldn't consider it good practice, but ok. I'm also not sure this will be optimized away, so there might be a (admittedly small) performance impact.

Reply:
Removed.